src/metabase/api/common.clj

```diff
@@ -255,6 +256,51 @@
                                   (ns-name *ns*) fn-name)))
       (assoc parsed :fn-name fn-name, :route route, :docstr docstr))))
 
+(def ^:private content-type->regex
+  "A map of content type to a regex that should match that content type."
+  {:content/json #"^application/(.+\+)?json"
+   :content/form #"(^application/x-www-form-urlencoded)|(multipart/form-data)|(application/form)"
+   :content/*    (constantly true)})
+
+(defn content-type-matches?
+  "Takes the request's content type and the allowable content types (:content/json, :content/form) and verifies if the actual content type is allowed."
+  [actual allowable]
+  (letfn [(allowable-content? [content-type]
+            (let [matcher (content-type->regex content-type)]
+              (if (instance? java.util.regex.Pattern matcher)
+                (when actual
+                  (re-find (content-type->regex content-type) actual))
+                (matcher actual))))]
+    (some allowable-content? allowable)))
+
+(defn make-route
+  "Create a route for our defendpoint. For non-POST methods, just returns the route. For POST methods, adds a second
+  check that the content type matches expected. By default this is \"application/json\", but can supply others in
+  metadata like :content/form for form data."
+  [route method content-types]
+  (if (= (str/lower-case (name method)) "post")
+    (do
+      (when (seq content-types)
+        (run! (fn [content-type]
+                (or (content-type->regex content-type)
+                    (throw (ex-info (trs "Unrecognized content type: {0}\n Valid: {1}"
+                                         content-type
+                                         (keys content-type->regex))
+                                    {}))))
+              content-types))
```

Comment:
Could you do this validation as part of the `s/fdef` for `defendpoint`? See if the method is the symbol `POST`, and if it is validate the metadata or something. Idk it might be quicker feedback than doing it this way